cmd/agent-actions/actions/delete_node_handler.go

```diff
@@ -0,0 +1,29 @@
+package actions
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+
+	"k8s.io/client-go/kubernetes"
+
+	"castai-agent/cmd/agent-actions/telemetry"
+)
+
+func newDeleteNodeHandler(clientset *kubernetes.Clientset) ActionHandler {
+	return &deleteNodeHandler{
+		clientset: clientset,
+	}
+}
+
+type deleteNodeHandler struct {
+	clientset *kubernetes.Clientset
```

Comment:
Should depend on the `kubernetes.Interface` as you'll want to be passing the fake clients in tests.